dev/SearchBox.js

```diff
@@ -1,6 +1,9 @@
 import React, {Component, PropTypes as T} from 'react';
 import connect from '../src/connect.js';
 
-export default connect()(function({helper}) {
-  return <input onChange={(e) => helper.setQuery(e.target.value).search()} />;
+export default connect()(function({helper, search}) {
```

Comment:
> Are you implementing default state so that reset still applies the default values?

Yep! The idea in instantsearch-react is to have two states: the user state, which is controlled via the URL/the helper provided in props/user interaction with the UI, and the default state, which is there for UI components that rely on some data always being fetched in order to display correctly. For instance, the RefinementList component requires the facet values corresponding to its `attributeName` prop to be in the search results.